wsd/DocumentBroker.cpp

```diff
@@ -1539,7 +1538,10 @@ std::size_t DocumentBroker::removeSession(const std::string& id)
         std::shared_ptr<ClientSession> session = it->second;
 
         // Last view going away, can destroy.
-        _markToDestroy = (_sessions.size() <= 1);
+        if (_sessions.size() <= 1)
+            _docState.markToDestroy();
+        else
+            assert(!_docState.isMarkedToDestroy());
```

Comment:
I am not fan at all of a pattern like "if (x) { blah(); } else assert(...) ;-)

Let's rather do a clear "assert(condition && other condiotion)" that is not a lone thing in an else branch please...

Reply:
Yes, I suppose that could be easier to read in some cases. Let me change it.